### PR TITLE
CreateAnalyzerResultCommand: Add the parameter `--scancode-version`

### PR DESCRIPTION
**Pull request by fviernau**
Allow to narrow down the matching of scan result entries in the postgres
storage to a given ScanCode version.

This enables the use case to re-scan a given set of package IDs which
have been scanned before with the specified ScanCode version.

### CODE REVIEW

File: helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt [hunk 1]
```diff
@@ -87,9 +87,14 @@ internal class CreateAnalyzerResultCommand : CliktCommand(
                 "-P ort.scanner.storages.postgres.schema=testSchema"
     ).associate()
 
+    private val scanCodeVersion by option(
+        "--scan-code-version",
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt:91]: How about naming this `--scancode-version`, like the `scancode` CLI?

File: helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt [hunk 2]
```diff
@@ -87,9 +87,14 @@ internal class CreateAnalyzerResultCommand : CliktCommand(
...
+        help = "The scan code version to match for equality against the scan results."
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt:92]: s/scan code/ScanCode/

Comment (sschuberth) [reply helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt:92]: Also "to match for equality against" sounds a bit clumsy... how about just "to match for in the scan results"?

File: helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt [hunk 3]
```diff
@@ -115,15 +120,20 @@ private data class ScannedPackage(
...
     val query = """
         SELECT 
             s.identifier as id,
             s.scan_result->>'provenance' as provenance,
             (s.scan_result->'summary'->>'start_time')::timestamp as start_time
         FROM 
             scan_results s 
-        WHERE 
+        WHERE
+            ${ scanCodeVersion?.let { "s.scan_result->'scanner'->>'version' = '$it' AND" }.orEmpty() }
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt:136]: Just an idea (typing blindly):

    val whereClause = listOfNotNull(
        scanCodeVersion?.let { "s.scan_result->'scanner'->>'version' = '$it'",
        "s.identifier = ANY(?)"
    ).joinToString(" AND ")

    // ....

    WHERE $whereClause;

File: helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt (hunk 1 above)
Comment (sschuberth) [reply helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt]: The commit message title still has `--scan-code-version`.

**Comment by sschuberth** [comment]
PS: A failing test will be fixed by my https://github.com/oss-review-toolkit/ort/pull/4743/commits/42f090d884e0e6b798968ac0d8ad9a5fa79f877e.

File: helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt [hunk 4]
```diff
@@ -115,16 +120,25 @@ private data class ScannedPackage(
...
 
-private fun getScannedPackages(connection: Connection, ids: Collection<Identifier>): List<ScannedPackage> {
+private fun getScannedPackages(
+    connection: Connection,
+    ids: Collection<Identifier>,
+    scanCodeVersion: String?
+): List<ScannedPackage> {
+    val whereClause = listOfNotNull(
+        "s.identifier = ANY(?)",
+        scanCodeVersion?.let { "s.scan_result->'scanner'->>'version' = '$it'" },
```
Comment (sschuberth) [inline helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt:130]: Nit: Please remove the trailing comma.

**Review by sschuberth — DISMISSED** [review]

**Review by sschuberth — APPROVED** [review]

**Review by sschuberth — CHANGES_REQUESTED** [review]
Is that the right commit on that branch? It seems to mismatch the PR title at least...

**Comment by fviernau** [comment]
> Is that the right commit on that branch? It seems to mismatch the PR title at least...

Good catch, it was not the right one.

**Review by sschuberth — APPROVED** [review]

**Comment by fviernau** [comment]
Merged despite failing unrelated tests.

**Pull request merged**